File: twitter/java-data-extraction-service/src/main/java/za/ac/wits/cpd/service/bigdatavisual/Tweet.java
package za.ac.wits.cpd.service.bigdatavisual;

import java.io.Serializable;
import java.util.GregorianCalendar;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlElementRef;
import javax.xml.bind.annotation.XmlRootElement;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * A simplified data structure for representing a Tweet.
 *
 * @author Matsobane Khwinana (Matsobane.Khwinana@momentum.co.za)
 */
@Data
@XmlRootElement
@NoArgsConstructor
@AllArgsConstructor
@XmlAccessorType(XmlAccessType.FIELD)
public class Tweet implements Serializable{
    private static final long serialVersionUID = 7042265067804136769L;
    
    @XmlAttribute
    private Long twitterId;
    @XmlAttribute
    private String text;
    @XmlElementRef
    private GeoLocation geoLocation;
    @XmlAttribute
    private String createdBy;
    @XmlAttribute
    private GregorianCalendar createdAt;
    @XmlAttribute
    private Long favouriteCount;
    @XmlAttribute
    private String hashtags[];
    @XmlAttribute
    private String inReplyToName;
    @XmlAttribute
    private Long inReplyToStatusId;
    @XmlAttribute
    private Long inReplyToUserId;
    @XmlAttribute
    private Long quotedStatusId;
    @XmlAttribute
    private boolean isRetweet;
    @XmlAttribute
    private boolean retweeted;
    @XmlAttribute
    private Long retweetedCount;
    @XmlAttribute
    private String language;
    @XmlAttribute
    private boolean sensitive;
    @XmlAttribute
    private String url;
}

File: spark/TwitConPro/src/main/scala-2.10/org/TwitConPro/JsonFormats/Tweet.scala
package org.TwitConPro.JsonFormats

import java.time.{Instant, ZonedDateTime}

/**
  * Created by Gareth on 2016/06/11.
  */
case class Tweet(//_id: Option[String],
                 //createdBy: String,
                 createdAt: Instant,
                 /*coords: Array[Either[String, Double]],
                 favouriteCount: Int,
                 hashtags: Array[Either[HashTag, String]],
                 twitterID: String,
                 inReplyToName: String,
                 inReplyToStatusID: Either[String, Int],
                 inReplyToUserID: Either[String, Int],
                 isRetweet: Boolean,
                 language: String,
                 place: String,
                 sensitive: Boolean,
                 quotedStatusID: Either[String, Int],
                 retweeted: Boolean,
                 retweetedCount: Int,*/
                 tweetText: String//,
                 //tweetURL: String
                )
